jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/AsyncAdHocSubprocessTest.java

```diff
@@ -91,6 +126,81 @@ public void testAdHocSubAsyncProcessAuto() throws Exception {
         assertProcessInstanceAborted(id);
     }
 
+    @Test(timeout = 30000)
+    public void testErrorMultilevelActivity() throws Exception {
+
+        
+        
+        KieSession ksession = createKSession(ERROR_MULTILEVEL_MAIN_PROCESS, ERROR_MULTILEVEL_CHILD_PROCESS, ERROR_MULTILEVEL_CHILD_CHILD_PROCESS, ERROR_MULTILEVEL_EXCEPTION_PROCESS);
+        final CountDownLatch latch = new CountDownLatch(1);
+        ksession.addEventListener(new DefaultProcessEventListener() {
+            @Override
+            public void afterProcessCompleted(ProcessCompletedEvent event) {
+                if("nested-abort.level1P".equals(event.getProcessInstance().getProcessId())) {
+                    latch.countDown();
+                }
+            }
+        });
+        ksession.getWorkItemManager().registerWorkItemHandler("Rest", new RestWorkItemHandler());
+        ProcessInstance pi = ksession.startProcess(ERROR_MULTILEVEL_MAIN_PROCESS_ID);
+        latch.await();
+
+        Thread.sleep(1000L);
+        assertThat(getLogService().findProcessInstance(pi.getId()).getStatus(), is(3));
```

Comment:
It might be clearer
```suggestion
        assertThat(getLogService().findProcessInstance(pi.getId()).getStatus(), is(ProcessInstance.STATE_ABORTED));
```

Reply:
ok